content/sensu-go/6.1/operations/maintain-sensu/troubleshoot.md

```diff
@@ -379,6 +379,123 @@ Since the `lsb_release` package is not installed, the agent will not be able to
+
+The etcd default maximum database size is 2 GB.
+If you suspect your etcd database has exceeded the maximum size, run this command to confirm cluster size:
+
+{{< code shell >}}
+# get current status with database size
+etcdctl endpoint status
+https://backend01:2379, 88db026f7feb72b4, 3.3.22, 2.1GB, false, 144, 18619245
+https://backend02:2379, e98ad7a888d16bd6, 3.3.22, 2.1GB, true, 144, 18619245
+https://backend03:2379, bc4e39432cbb36d, 3.3.22, 2.1GB, false, 144, 18619245
+{{< /code >}}
+
+To restore an etcd cluster with a database size that exceeds 2 GB, run these commands:
+
+{{< code shell >}}
+# get current revision number
+etcdctl endpoint status --write-out="json" | egrep -o '"revision":[0-9]*' | egrep -o '[0-9].*'
+
+# compact to revision, substitute revision obtained above for $rev
+etcdctl compact $rev
+
+# defrag to free space
+etcdctl defrag
+
+# confirm effective
+etcdctl endpoint status
+https://backend01:2379, 88db026f7feb72b4, 3.3.22, 1.0 MB, false, 144, 18619245
+https://backend02:2379, e98ad7a888d16bd6, 3.3.22, 1.0 MB, true, 144, 18619245
+https://backend03:2379, bc4e39432cbb36d, 3.3.22, 1.0 MB, false, 144, 18619245
+{{< /code >}}
+
+## Datastore performance
+
+### Overview
+
+In a default deployment, Sensu uses [etcd datastore][17] for both configuration and state. As the number of checks and entities in your Sensu installation increases, so does the volume of read and write requests to etcd database.
+
+One trade-off in etcd's design is it's sensitivity to disk and CPU latency. When certain latency tollerances are regularly exceeded, failures will cascade. Sensu will attempt to recover from these conditions when it can, but 
+
+To maximize Sensu Go performance, we recommend that you:
+ * Follow our [recommended backend hardware requirements][17].
+ * Immplement [documented etcd system tuning practices][14].
+ * [Benchmark your etcd storage volume][15] to establish baseline IOPS for your system.
+ * [Scale event storage using Postgres][16], reducing the overall volume of etcd transactions.
+
+ As your Sensu deployments grow, prevention against the issues associated with poor datastore performance relies on ongoing collection and review of [Sensu time-series performance metrics][18].
+
+### Symptoms of poor performance
+
+At the default "warn" log level, you may see messages like these from your Sensu backend:
+
+{{< code json >}}
+{"component":"etcd","level":"warning","msg":"read-only range request \"key:\\\"/sensu.io/handlers/default/keepalive\\\" limit:1 \" with result \"range_response_count:0 size:6\" took too long (169.767546ms) to execute","pkg":"etcdserver","time":"..."}
+{{< /code >}}
+
+The above message indicates that a database query ("read-only range request") exceeded a 100 millisecond threshold hard-coded into etcd. Messages like these are helpful in so far as they can alert us to a trend, but occasional warnings of this nature aren't necessarily indicative of a problem.
```

Comment:
```suggestion
The above message indicates that a database query ("read-only range request") exceeded a 100-millisecond threshold hard-coded into etcd.
Messages like these are helpful because they can alert you to a trend, but these occasional warnings don't necessarily indicate a problem.
```